src/k2/cpo/service/HealthMonitor.h

```diff
@@ -102,6 +103,10 @@ class HealthMonitor {
     // required for seastar::distributed interface
     seastar::future<> gracefulStop();
     seastar::future<> start();
+
+    std::vector<String> getNodepoolEndpoints();
```

Comment:
return const& instead?

Reply:
Use invoke_on internally

Reply:
invoke_on requires a member function so I couldn't use a lambda, but the helper functions are private